service/worker/scanner/executions/task.go

```diff
@@ -208,14 +211,25 @@ func (t *task) handleFailures(
 		case mutableStateRetentionFailureType:
 			executionInfo := mutableState.GetExecutionInfo()
 			runID := mutableState.GetExecutionState().GetRunId()
-			_, err := t.historyClient.DeleteWorkflowExecution(t.ctx, &historyservice.DeleteWorkflowExecutionRequest{
-				NamespaceId: executionInfo.GetNamespaceId(),
-				WorkflowExecution: &commonpb.WorkflowExecution{
+			ns, err := t.registry.GetNamespaceByID(namespace.ID(executionInfo.GetNamespaceId()))
+			switch err.(type) {
+			case *serviceerror.NotFound,
+				*serviceerror.NamespaceNotFound:
+				t.logger.Error("Garbage data in DB after namespace is deleted", tag.WorkflowNamespaceID(executionInfo.GetNamespaceId()))
+				// We cannot do much in this case. It just ignores this error.
```

Comment:
hmmm, looks like we still need a way to clean up in this case. Admin DeleteWorkflowExecution should probably take in namespaceID instead of namespaceName?

Maybe create a task for this?